apisix/core/string.lua

```diff
@@ -78,4 +78,12 @@ function _M.rfind_char(s, ch, idx)
 end
 
 
+-- reduce network consumption by compressing string indentation
```

Comment:
I think the compress way is not ideal as it might also change the literal string (spaces after `,`), so it should be used carefully, add some comments to give a hint will be better.